chain/beacon/sync_manager.go

```diff
@@ -415,6 +415,11 @@ func (s *SyncManager) tryNode(global context.Context, from, upTo uint64, peer ne
 				}
 			} else {
 				if err := s.store.Put(cnode, beacon); err != nil {
+					if errors.Is(err, ErrBeaconAlreadyStored) {
+						logger.Debugw("Put: race with aggregation", "with_peer", peer.Address(), "err", err)
+						return true
```

Comment:
Actually we might want to compare the value with upTo before returning true. if `beacon.Round == upTo` then `true` but otherwise false... (Could happen in catchupmode that we're lagging behind but the aggregator still raced the sync manager)

Reply:
Good catch.